### PR TITLE
orderBy can take a Function as its first argument

### PR DESCRIPTION
**Pull request by posva**
Hello,

This feature was asked in #2518. The point of this generic filter is to support any kind of sorting by using a custom comparator. Tell me your thoughts on the feature and the code :smile: 

PS: I did the PR instead of the issue because it took little to no time to do it and I didn't have internet when I coded it :stuck_out_tongue:

### CODE REVIEW

File: src/filters/array-filters.js [hunk 1]
```diff
@@ -110,6 +110,27 @@ export function orderBy (arr, sortKeys, reverse) {
 }
 
 /**
+ * Filter filter for arrays
+ *
+ * @param {Function} [compare]
```
Comment (fullfs) [inline src/filters/array-filters.js]: `@param {Array} [arr]` is missing

**Comment by yyx990803** [comment]
Forgot to comment in the feature request issue, but I think this should just be part of `orderBy`. The first argument can either be a string or a function, this is how `filterBy` works currently.

**Comment by posva** [comment]
Ok, so instead of giving a String or an Array of string you can give your own Function.

**Comment by posva** [comment]
Let me review this a last time tomorrow in the morning (I want to be fresh :smile: )
I'll also update the doc

**Comment by posva** [comment]
ping @yyx990803

**Pull request merged**